src/site/artifacts/cargo_dist.rs

```diff
@@ -21,25 +22,40 @@ pub fn get_os(name: &str) -> Option<&str> {
 }
 
 pub fn fetch_manifest(config: &Config) -> Result<DistManifest> {
-    let url = create_download_link(config, &String::from("dist-manifest.json"));
+    if let Some(repo) = &config.repository {
+        let releases = fetch_releases(repo.as_str())?;
+        let first = releases
+            .iter()
+            .find(|release| !release.prerelease)
+            .unwrap_or(&releases[0]);
+        let url = create_download_link(
+            config,
+            &String::from("dist-manifest.json"),
+            Some(first.tag_name.to_owned()),
+        );
 
-    match reqwest::blocking::get(&url)?.error_for_status() {
-        Ok(resp) => match resp.json::<DistManifest>() {
-            Ok(manifest) => Ok(manifest),
-            Err(e) => Err(OrandaError::CargoDistManifestParseError {
+        match reqwest::blocking::get(&url)?.error_for_status() {
```

Comment:
we should add a blocking api to axo asset because we really should be using it for these fetches because of the error handling

Reply:
(i am the one who added the errors heh hey but yeah)